server/config/config.go

```diff
@@ -91,6 +91,17 @@ type GRPCConfig struct {
 	Address string `mapstructure:"address"`
 }
 
+// StateSyncConfig defines the state sync snapshot configuration.
+type StateSyncConfig struct {
+	// SnapshotInterval sets the interval at which state sync snapshots are taken.
+	// 0 disables snapshots. Must be a multiple of PruningKeepEvery.
```

Comment:
Will this change in tandem with the pruning settings? If I stop my node change the pruning strategy then restart without changing this field what will happen?

Reply:
You'll get an error. The snapshot interval must be a multiple of the pruning keepever interval.

This is a somewhat arbitrary limitation. Snapshotting just needs a guarantee that heights won't be removed while being snapshotted, and this is currently the only way the SDK can give this guarantee. If the snapshotter could request the pruner to delay pruning of a height until the snapshot is complete then these settings no longer need to match.

Reply:
Let's create a followup issue for that @erikgrinaker. This isn't a blocker and what you have now should suffice in most cases.

Reply:
Added #7188 for this.